src/lib.rs

```diff
@@ -118,8 +118,36 @@ impl From<CRSError> for BackendError {
 }
 
 /// The number of bytes necessary to store a `FieldElement`.
+///
+/// Barretenberg uses the bn254 elliptic curve, it then requires
+/// 32 bytes (256 bits) to represent each `FieldElement`
 const FIELD_BYTES: usize = 32;
 
+// These tests ensure that some of our assumptions about `FieldElement` hold true.
+// i.e. it can be represented in `FIELD_BYTES` bytes and can be converted to a byte array of that length.
+#[cfg(test)]
+mod field_size_checks {
+    use acvm::FieldElement;
+
+    use crate::FIELD_BYTES;
+
+    #[test]
+    fn check_field_size() {
+        assert_eq!(FIELD_BYTES, FieldElement::max_num_bytes() as usize);
```

Comment:
Is there a reason we aren't using `FieldElement::max_num_bytes()` within our code instead of setting the FIELD_BYTES constant?

Reply:
A couple of reasons:
- `FieldElement::max_num_bytes()` is much more verbose than `FIELD_BYTES`
- `FieldElement::max_num_bytes()` is a `u32` whereas a lot of the usage of `FIELD_BYTES` requires a `usize` so we need to do casting in function signatures which is yucky.